finetuner/tuner/callback/training_checkpoint.py

```diff
@@ -0,0 +1,166 @@
+import os
+import pickle
+from typing import TYPE_CHECKING
+import shutil
+
+import keras
+import paddle
+import torch
+from jina.logging.logger import JinaLogger
+
+from finetuner.helper import get_framework
+
+from .base import BaseCallback
+
+if TYPE_CHECKING:
+    from ..base import BaseTuner
+
+
+class TrainingCheckpoint(BaseCallback):
+    """
+    Callback to save model at every epoch or the last K epochs
+    """
+
+    def __init__(self, save_dir: str, last_k_epochs: int = None):
+        """
+        :param save_dir: string, path to save the model file.
+        :param last_k_epochs: this parameter is an integer. It allows you to save
+            on only the most recent epochs and not on every epoch.
+            For example if `last_k_epochs = 3` and the total number of epochs is 10
+            epoch 5 -> saved 5, 4, 3
+            epoch 6 -> saved 6, 5, 4 (here saved epoch 3 is deleted)
+            epoch 7 -> saved 7, 6, 5 (here saved epoch 4 is deleted)
```

Comment:
I think this is not needed. Just say that as training progresses, only the most recenkt `k` checkpoints will be kept, and older ones removed each epoch